spec/components/custom_wizard/mapper_spec.rb

```diff
@@ -250,4 +274,75 @@
       user: user1
     ).perform).to eq(false)
   end
+
+  context "Output Templating" do
+    it "passes the correct values to the template" do
+      template = "w{step_1_field_1}"
+      result = template_mapper.interpolate(
+        template.dup,
+        template: true,
+        user: true,
+        wizard: true,
+        value: true
+      )
+      expect(result).to eq(template_params["step_1_field_1"])
+    end
+
+    it "treats replaced values as string literals" do
+      template = '{{ "w{step_1_field_1}" | size }}'
+      result = template_mapper.interpolate(
+        template.dup,
+        template: true,
+        user: true,
+        wizard: true,
+        value: true
+      )
+      expect(result).to eq(template_params["step_1_field_1"].size.to_s)
+    end
+
+    it "allows the wizard values to be used inside conditionals" do
+      template = <<-LIQUID
+        {%- if "w{step_1_field_1}" contains "ello" -%}
+          Correct
+        {%- else -%}
+          Incorrect
+        {%-endif-%}
+      LIQUID
+      result = template_mapper.interpolate(
+        template.dup,
+        template: true,
+        user: true,
+        wizard: true,
+        value: true
+      )
+      expect(result).to eq("Correct")
+    end
+
+    it "can access data passed to render method as variable" do
+      template = "{{step_1_field_1.size}}"
+      result = template_mapper.interpolate(
+        template.dup,
+        template: true,
+        user: true,
+        wizard: true,
+        value: true
+      )
+      expect(result).to eq(template_params["step_1_field_1"].size.to_s)
+    end
+
+    it "custom filter: 'first_non_empty' gives first non empty element from list" do
```

Comment:
@fzngagan We need a test for when all elements are empty. Also for when multiple elements are not empty. Currently we're making the test too easy, i.e. there's only one non-empty and it's at the end.

Reply:
Added a few more tests to ensure its doing the right thing